{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<h3><a href=\"https://pollytikhonova.github.io/bioinformatics_homeworks/Medicine_modelling\"><img src=\"https://pollytikhonova.github.io/bioinformatics_homeworks/GitHub-Mark-32px.png\" alt=\"Вернуться в репозиторий\" title=\"Вернуться в репозиторий\" style=\"max-width:100%;  display: inline-block; margin-right: 10px;\"></a>\n",
    "Tikhonova Polina. Homework 6. </h3>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sys \n",
    "import modeller \n",
    "import _modeller\n",
    "import modeller.automodel \n",
    "import nglview"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "                         MODELLER 9.19, 2017/07/19, r11078\n",
      "\n",
      "     PROTEIN STRUCTURE MODELLING BY SATISFACTION OF SPATIAL RESTRAINTS\n",
      "\n",
      "\n",
      "                     Copyright(c) 1989-2017 Andrej Sali\n",
      "                            All Rights Reserved\n",
      "\n",
      "                             Written by A. Sali\n",
      "                               with help from\n",
      "              B. Webb, M.S. Madhusudhan, M-Y. Shen, G.Q. Dong,\n",
      "          M.A. Marti-Renom, N. Eswar, F. Alber, M. Topf, B. Oliva,\n",
      "             A. Fiser, R. Sanchez, B. Yerkovich, A. Badretdinov,\n",
      "                     F. Melo, J.P. Overington, E. Feyfant\n",
      "                 University of California, San Francisco, USA\n",
      "                    Rockefeller University, New York, USA\n",
      "                      Harvard University, Cambridge, USA\n",
      "                   Imperial Cancer Research Fund, London, UK\n",
      "              Birkbeck College, University of London, London, UK\n",
      "\n",
      "\n",
      "Kind, OS, HostName, Kernel, Processor: 4, Linux shadbox 3.2.0-29-generic x86_64\n",
      "Date and time of compilation         : 2017/07/19 14:40:43\n",
      "MODELLER executable type             : x86_64-intel8\n",
      "Job starting time (YY/MM/DD HH:MM:SS): 2017/12/02 23:53:11\n",
      "\n"
     ]
    }
   ],
   "source": [
    "env=modeller.environ()\n",
    "env.io.hetatm=True"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Взяла белок ли́сого кузу́ (или щёткохвост, или лисовидный поссум, или обыкновенный кузу-лиса — млекопитающее семейства кускусовых).\n",
    "<img src=\"possum.jpg\">"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "! wget http://www.pdb.org/pdb/files/1lmp.pdb"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "! wget http://www.uniprot.org/uniprot/P51782.fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "alignm=modeller.alignment(env)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "read_pd_459W> Residue type  NAG not recognized. 'automodel' model building\n",
      "              will treat this residue as a rigid body.\n",
      "              To use real parameters, add the residue type to ${LIB}/restyp.lib,\n",
      "              its topology to ${LIB}/top_*.lib, and suitable forcefield\n",
      "              parameters to ${LIB}/par.lib.\n",
      "rdpdb___459W> Residue type  NDG not recognized. 'automodel' model building\n",
      "              will treat this residue as a rigid body.\n",
      "              To use real parameters, add the residue type to ${LIB}/restyp.lib,\n",
      "              its topology to ${LIB}/top_*.lib, and suitable forcefield\n",
      "              parameters to ${LIB}/par.lib.\n"
     ]
    }
   ],
   "source": [
    "alignm.append(file='P51782.fasta', align_codes='all',alignment_format='FASTA')\n",
    "## создадим модель\n",
    "mdl = modeller.model(env, file='1lmp.pdb', model_segment=('FIRST:'+'A', 'LAST:'+'A'))\n",
    "## и добавим в выравнивание\n",
    "alignm.append_model(mdl, atom_files='1lmp.pdb', align_codes='1lmp')\n",
    "alignm[0].code = 'without_ligand'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "SALIGN_____> adding the next group to the alignment; iteration    1\n"
     ]
    }
   ],
   "source": [
    "alignm.salign()\n",
    "alignm.write(file='all_in_one.ali', alignment_format='PIR')\n",
    "alignm[0].code = 'without_ligand'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "without_ligand 1lmp\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "read_to_681_> topology.submodel read from topology file:        3\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "patch_s_522_> Number of disulfides patched in MODEL:        4\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  C1 -->  CT2\n",
      "              This message is written only for the first such atom.\n",
      "0 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "0 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    13014    11982\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      147\n",
      "Number of all, selected real atoms                :     1179    1179\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    11982   11982\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2358\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         790.6201\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1205       0      0   0.006   0.006      12.310       1.000\n",
      " 2 Bond angle potential               :    1624       0      5   1.969   1.969      127.56       1.000\n",
      " 3 Stereochemical cosine torsion poten:     773       0     36  48.801  48.801      280.90       1.000\n",
      " 4 Stereochemical improper torsion pot:     506       0      0   1.244   1.244      18.239       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2358       0      0   0.001   0.001     0.54846       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    2380       0      3   0.146   0.146      42.705       1.000\n",
      "10 Distance restraints 2 (N-O)        :    2535       0      0   0.183   0.183      73.640       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     146       0      1   4.219   4.219      30.652       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     125       0      0  61.593  61.593      8.5635       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:      98       0      0  75.037  75.037      37.092       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      40       0      1  84.580  84.580      28.607       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      19       0      0  81.349  81.349      11.549       1.000\n",
      "18 Disulfide distance restraints      :       4       0      0   0.006   0.006     0.27971E-01   1.000\n",
      "19 Disulfide angle restraints         :       8       0      0   1.711   1.711     0.51698       1.000\n",
      "20 Disulfide dihedral angle restraints:       4       0      0  28.648  28.648      3.2186       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    1577       0      0   0.395   0.395      23.522       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     145      16     20  22.136  57.429      44.195       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     793       0      0   0.725   0.725      46.776       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: without_ligand.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   14269.7188\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4127   3V   4L C   N      23   25 -118.82 -108.50   44.55    2.29  -63.50  153.29   24.71\n",
      "    1          4L   4L N   CA     25   26  175.84  132.50                  -41.20\n",
      "    2   4129   5L   6L C   N      39   41 -152.04 -108.50   43.82    2.15  -63.50 -169.46   23.34\n",
      "    2          6L   6L N   CA     41   42  127.52  132.50                  -41.20\n",
      "    3   4130   6L   7L C   N      47   49 -133.26 -108.50   28.57    1.54  -63.50  174.05   21.41\n",
      "    3          7L   7L N   CA     49   50  118.25  132.50                  -41.20\n",
      "    4   4131   7L   8G C   N      55   57   69.34   78.70   15.64    0.79   82.20  173.34    7.76\n",
      "    4          8G   8G N   CA     57   58 -178.63 -166.10                    8.50\n",
      "    5   4132   8G   9F C   N      59   61 -132.74 -124.20   12.89    0.36  -63.20  176.98   29.11\n",
      "    5          9F   9F N   CA     61   62  152.95  143.30                  -44.30\n",
      "    6   4134  10I  11F C   N      78   80  -99.40 -124.20   40.70    1.19  -63.20  159.49   19.86\n",
      "    6         11F  11F N   CA     80   81  111.03  143.30                  -44.30\n",
      "    7   4136  12C  13S C   N      95   97  -78.33  -72.40   14.05    0.66  -64.10  160.49   12.53\n",
      "    7         13S  13S N   CA     97   98  165.14  152.40                  -35.00\n",
      "    8   4137  13S  14M C   N     101  103  -72.52  -73.00    6.59    0.45  -63.40  170.17   26.12\n",
      "    8         14M  14M N   CA    103  104  149.57  143.00                  -40.50\n",
      "    9   4138  14M  15A C   N     109  111 -125.52 -134.00   10.68    0.59  -62.50  177.20   32.41\n",
      "    9         15A  15A N   CA    111  112  153.49  147.00                  -40.90\n",
      "   10   4139  15A  16A C   N     114  116 -135.89 -134.00   12.49    0.66  -62.50  175.81   32.74\n",
      "   10         16A  16A N   CA    116  117  159.34  147.00                  -40.90\n",
      "   11   4140  16A  17H C   N     119  121 -107.99 -125.60   21.78    0.97  -63.20  172.02   24.43\n",
      "   11         17H  17H N   CA    121  122  151.62  138.80                  -42.30\n",
      "   12   4141  17H  18G C   N     129  131  128.64   78.70   52.88    2.14   82.20  174.31   10.70\n",
      "   12         18G  18G N   CA    131  132  176.51 -166.10                    8.50\n",
      "   13   4142  18G  19K C   N     133  135  -68.65  -70.20   20.62    1.46  -62.90  160.74   20.57\n",
      "   13         19K  19K N   CA    135  136  119.84  140.40                  -40.80\n",
      "   14   4177  53E  54S C   N     436  438 -137.47  -64.10   81.11    8.55  -64.10   81.11    8.55\n",
      "   14         54S  54S N   CA    438  439   -0.42  -35.00                  -35.00\n",
      "   15   4188  64N  65P C   N     521  523  -73.93  -64.50    9.81    1.08  -58.70  175.66   13.32\n",
      "   15         65P  65P N   CA    523  524  144.50  147.20                  -30.50\n",
      "   16   4241 117I 118V C   N     932  934  -61.42  -62.40   51.11    6.53  -62.40   51.11    6.53\n",
      "   16        118V 118V N   CA    934  935    8.71  -42.40                  -42.40\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    0    3    8   45   75  116   98  129  155  171  183\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      147\n",
      "Number of all, selected real atoms                :     1179    1179\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    11982   11982\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2410\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         890.2857\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1205       0      0   0.006   0.006      12.352       1.000\n",
      " 2 Bond angle potential               :    1624       0      6   2.171   2.171      152.58       1.000\n",
      " 3 Stereochemical cosine torsion poten:     773       0     35  49.119  49.119      287.36       1.000\n",
      " 4 Stereochemical improper torsion pot:     506       0      1   1.584   1.584      26.643       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2410       0      0   0.002   0.002     0.88394       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    2380       0      2   0.166   0.166      57.012       1.000\n",
      "10 Distance restraints 2 (N-O)        :    2535       0      0   0.185   0.185      75.602       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     146       0      1   4.287   4.287      31.647       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     125       0      0  68.796  68.796      24.181       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:      98       0      0  70.106  70.106      36.112       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      40       0      0  79.528  79.528      31.384       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      19       0      0  90.345  90.345      12.927       1.000\n",
      "18 Disulfide distance restraints      :       4       0      0   0.007   0.007     0.36999E-01   1.000\n",
      "19 Disulfide angle restraints         :       8       0      0   1.723   1.723     0.52438       1.000\n",
      "20 Disulfide dihedral angle restraints:       4       0      0  29.770  29.770      3.3614       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    1577       0      0   0.438   0.438      27.315       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     145      17     18  24.303  60.966      53.991       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     793       0      1   0.705   0.705      56.375       1.000\n",
      "27 Distance restraints 5 (X-Y)        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: without_ligand.V99990002\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   15682.0039\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4125   1M   2K C   N       7    9  -63.80  -70.20    8.60    0.72  -62.90  173.05   22.52\n",
      "    1          2K   2K N   CA      9   10  146.15  140.40                  -40.80\n",
      "    2   4127   3V   4L C   N      23   25  -92.58 -108.50   16.60    0.85  -63.50 -179.25   23.59\n",
      "    2          4L   4L N   CA     25   26  137.19  132.50                  -41.20\n",
      "    3   4128   4L   5L C   N      31   33 -113.70 -108.50   27.66    1.43  -63.50  166.87   26.27\n",
      "    3          5L   5L N   CA     33   34  159.66  132.50                  -41.20\n",
      "    4   4129   5L   6L C   N      39   41 -122.42 -108.50   14.54    0.65  -63.50 -172.61   23.39\n",
      "    4          6L   6L N   CA     41   42  136.69  132.50                  -41.20\n",
      "    5   4130   6L   7L C   N      47   49  -93.57  -70.70   27.91    1.91  -63.50  163.98   24.59\n",
      "    5          7L   7L N   CA     49   50  157.60  141.60                  -41.20\n",
      "    6   4131   7L   8G C   N      55   57  142.51 -167.20   50.98    0.80   82.20 -175.41   11.82\n",
      "    6          8G   8G N   CA     57   58 -177.04  174.60                    8.50\n",
      "    7   4132   8G   9F C   N      59   61  -88.75  -71.40   75.80    6.51  -63.20  114.12   14.23\n",
      "    7          9F   9F N   CA     61   62   66.92  140.70                  -44.30\n",
      "    8   4134  10I  11F C   N      78   80  -92.52  -71.40   53.32    4.70  -63.20  139.16   17.46\n",
      "    8         11F  11F N   CA     80   81   91.74  140.70                  -44.30\n",
      "    9   4136  12C  13S C   N      95   97 -160.88 -136.60   34.52    1.12  -64.10  177.89   19.87\n",
      "    9         13S  13S N   CA     97   98  175.73  151.20                  -35.00\n",
      "   10   4137  13S  14M C   N     101  103  -74.40  -73.00   20.38    1.35  -63.40  156.56   24.24\n",
      "   10         14M  14M N   CA    103  104  163.33  143.00                  -40.50\n",
      "   11   4138  14M  15A C   N     109  111  -59.72  -68.20   37.16    2.70  -62.50  150.05   24.73\n",
      "   11         15A  15A N   CA    111  112  109.13  145.30                  -40.90\n",
      "   12   4139  15A  16A C   N     114  116  152.02 -134.00   80.70    1.84  -62.50 -158.18   40.41\n",
      "   12         16A  16A N   CA    116  117  179.23  147.00                  -40.90\n",
      "   13   4140  16A  17H C   N     119  121   70.70   56.30   95.74    9.74  -63.20 -137.46   36.25\n",
      "   13         17H  17H N   CA    121  122  135.45   40.80                  -42.30\n",
      "   14   4141  17H  18G C   N     129  131  176.24 -167.20   16.92    0.26   82.20 -166.11   13.63\n",
      "   14         18G  18G N   CA    131  132  178.06  174.60                    8.50\n",
      "   15   4142  18G  19K C   N     133  135  -91.47  -70.20   28.83    2.41  -62.90  164.24   19.94\n",
      "   15         19K  19K N   CA    135  136  120.94  140.40                  -40.80\n",
      "   16   4177  53E  54S C   N     436  438 -137.42  -64.10   80.90    8.55  -64.10   80.90    8.55\n",
      "   16         54S  54S N   CA    438  439   -0.81  -35.00                  -35.00\n",
      "   17   4189  65P  66G C   N     528  530  -68.85  -62.40    7.73    1.46   82.20  160.40   12.23\n",
      "   17         66G  66G N   CA    530  531  -45.46  -41.20                    8.50\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    1    4   13   49   70  112  115  128  154  201  199\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "without_ligand.B99990001.pdb      790.62006\n",
      "without_ligand.B99990002.pdb      890.28571\n",
      "\n"
     ]
    }
   ],
   "source": [
    "s = alignm[0]\n",
    "pdb = alignm[1]\n",
    "\n",
    "print s.code, pdb.code\n",
    "\n",
    "## Создаем объект automodel\n",
    "a = modeller.automodel.automodel(env, alnfile='all_in_one.ali', knowns= pdb.code , sequence = s.code )\n",
    "\n",
    "a.name='mod'+s.code\n",
    "a.starting_model = 1\n",
    "a.ending_model = 2\n",
    "a.make()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "448efb3366a948bfb0a2aaaa49854769",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>NGLWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in Jupyter Notebook or JupyterLab, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another notebook frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nglview.show_structure_file('without_ligand.B99990001.pdb')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "910829b15e9342798071b6b51bfe7427",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>NGLWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in Jupyter Notebook or JupyterLab, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another notebook frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# original structure\n",
    "nglview.show_structure_file('1lmp.pdb')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Find ligand code"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " LYS  VAL  TYR  ASP  ARG  CYS  GLU  LEU  ALA  ARG  ALA  LEU  LYS  ALA  SER  GLY  MET  ASP  GLY  TYR \n",
      "  K    V    Y    D    R    C    E    L    A    R    A    L    K    A    S    G    M    D    G    Y  \n",
      " ALA  GLY  ASN  SER  LEU  PRO  ASN  TRP  VAL  CYS  LEU  SER  LYS  TRP  GLU  SER  SER  TYR  ASN  THR \n",
      "  A    G    N    S    L    P    N    W    V    C    L    S    K    W    E    S    S    Y    N    T  \n",
      " GLN  ALA  THR  ASN  ARG  ASN  THR  ASP  GLY  SER  THR  ASP  TYR  GLY  ILE  PHE  GLN  ILE  ASN  SER \n",
      "  Q    A    T    N    R    N    T    D    G    S    T    D    Y    G    I    F    Q    I    N    S  \n",
      " ARG  TYR  TRP  CYS  ASP  ASP  GLY  ARG  THR  PRO  GLY  ALA  LYS  ASN  VAL  CYS  GLY  ILE  ARG  CYS \n",
      "  R    Y    W    C    D    D    G    R    T    P    G    A    K    N    V    C    G    I    R    C  \n",
      " SER  GLN  LEU  LEU  THR  ASP  ASP  LEU  THR  VAL  ALA  ILE  ARG  CYS  ALA  LYS  ARG  VAL  VAL  LEU \n",
      "  S    Q    L    L    T    D    D    L    T    V    A    I    R    C    A    K    R    V    V    L  \n",
      " ASP  PRO  ASN  GLY  ILE  GLY  ALA  TRP  VAL  ALA  TRP  ARG  LEU  HIS  CYS  GLN  ASN  GLN  ASP  LEU \n",
      "  D    P    N    G    I    G    A    W    V    A    W    R    L    H    C    Q    N    Q    D    L  \n",
      " ARG  SER  TYR  VAL  ALA  GLY  CYS  GLY  VAL  NAG  NAG  NDG \n",
      "  R    S    Y    V    A    G    C    G    V    .    .    .  \n"
     ]
    }
   ],
   "source": [
    "n=20\n",
    "for i in range(0,len(alignm[1].residues),n):\n",
    "    if (i+n) < len(alignm[1].residues):\n",
    "        print str('{:^5}'*n).format(*[k.name for k in alignm[1].residues[i:i+n]])\n",
    "        print str('{:^5}'*n).format(*[k.code for k in alignm[1].residues[i:i+n]])\n",
    "    else:\n",
    "        n = len(alignm[1].residues)-i\n",
    "        print str('{:^5}'*n).format(*[k.name for k in alignm[1].residues[i:len(alignm[1].residues)]])\n",
    "        print str('{:^5}'*n).format(*[k.code for k in alignm[1].residues[i:len(alignm[1].residues)]])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "alignm2=modeller.alignment(env)\n",
    "residues = ''.join([k.code for k in alignm[0].residues]) + ''.join([k.code for k in alignm[1].residues[-3:]])\n",
    "alignm2.append_sequence(residues)\n",
    "## создадим модель\n",
    "mdl = modeller.model(env, file='1lmp.pdb', model_segment=('FIRST:'+'A', 'LAST:'+'A'))\n",
    "## и добавим в выравнивание\n",
    "alignm2.append_model(mdl, atom_files='1lmp.pdb', align_codes='1lmp')\n",
    "alignm2[0].code = 'with_ligand'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "SALIGN_____> adding the next group to the alignment; iteration    1\n",
      "with_ligand 1lmp\n",
      "automodel__W> Topology and/or parameter libraries already in memory. These will\n",
      "                be used instead of the automodel defaults. If this is not what you\n",
      "                want, clear them before creating the automodel object with\n",
      "                env.libs.topology.clear() and env.libs.parameters.clear()\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "\n",
      "getf_______W> RTF restraint not found in the atoms list:\n",
      "              residue type, indices:    10   147\n",
      "              atom names           : C     +N\n",
      "              atom indices         :  1177     0\n",
      "\n",
      "getf_______W> RTF restraint not found in the atoms list:\n",
      "              residue type, indices:    10   147\n",
      "              atom names           : C     CA    +N    O\n",
      "              atom indices         :  1177  1172     0  1178\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "patch_s_522_> Number of disulfides patched in MODEL:        4\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  C1 -->  CT2\n",
      "              This message is written only for the first such atom.\n",
      "43 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "43 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    14413    13381\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      150\n",
      "Number of all, selected real atoms                :     1222    1222\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    13381   13381\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2608\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         986.9169\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1205       0      0   0.007   0.007      17.018       1.000\n",
      " 2 Bond angle potential               :    1624       0      6   2.201   2.201      157.57       1.000\n",
      " 3 Stereochemical cosine torsion poten:     773       0     36  49.331  49.331      288.41       1.000\n",
      " 4 Stereochemical improper torsion pot:     506       0      1   1.544   1.544      25.815       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2608       2      2   0.008   0.008      17.726       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    2380       0      0   0.134   0.134      42.382       1.000\n",
      "10 Distance restraints 2 (N-O)        :    2535       0      0   0.180   0.180      77.130       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     146       0      1   4.076   4.076      28.608       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     125       0      4  77.022  77.022      40.143       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:      98       0      2  83.185  83.185      50.228       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      40       0      0  82.143  82.143      29.844       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      19       0      0  88.836  88.836      12.418       1.000\n",
      "18 Disulfide distance restraints      :       4       0      0   0.015   0.015     0.15137       1.000\n",
      "19 Disulfide angle restraints         :       8       0      0   2.254   2.254     0.89779       1.000\n",
      "20 Disulfide dihedral angle restraints:       4       0      0  23.797  23.797      2.2068       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    1577       0      0   0.426   0.426      44.445       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     145      18     18  22.873  62.767      54.413       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     793       0      1   0.777   0.777      76.270       1.000\n",
      "27 Distance restraints 5 (X-Y)        :    1399       0      0   0.042   0.042      21.236       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: with_ligand.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   17066.4570\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4125   1M   2K C   N       7    9  -82.94  -70.20   16.36    0.99  -62.90  169.72   23.19\n",
      "    1          2K   2K N   CA      9   10  150.67  140.40                  -40.80\n",
      "    2   4127   3V   4L C   N      23   25  -71.22  -70.70    4.60    0.33  -63.50  172.80   24.44\n",
      "    2          4L   4L N   CA     25   26  146.17  141.60                  -41.20\n",
      "    3   4128   4L   5L C   N      31   33  -66.46  -70.70    4.67    0.47  -63.50  175.26   24.50\n",
      "    3          5L   5L N   CA     33   34  143.56  141.60                  -41.20\n",
      "    4   4129   5L   6L C   N      39   41  -68.34  -70.70    3.34    0.21  -63.50  179.63   25.22\n",
      "    4          6L   6L N   CA     41   42  139.23  141.60                  -41.20\n",
      "    5   4130   6L   7L C   N      47   49  -74.07  -70.70    3.42    0.29  -63.50  176.95   25.19\n",
      "    5          7L   7L N   CA     49   50  142.17  141.60                  -41.20\n",
      "    6   4131   7L   8G C   N      55   57  130.25   78.70   52.50    2.38   82.20 -178.09   11.15\n",
      "    6          8G   8G N   CA     57   58 -176.04 -166.10                    8.50\n",
      "    7   4132   8G   9F C   N      59   61  -75.74  -71.40    4.34    0.29  -63.20  175.50   25.21\n",
      "    7          9F   9F N   CA     61   62  140.65  140.70                  -44.30\n",
      "    8   4134  10I  11F C   N      78   80 -129.14 -124.20    5.80    0.15  -63.20 -178.26   29.57\n",
      "    8         11F  11F N   CA     80   81  146.35  143.30                  -44.30\n",
      "    9   4136  12C  13S C   N      95   97  -68.42  -72.40   25.93    1.39  -64.10  161.84   11.48\n",
      "    9         13S  13S N   CA     97   98  126.78  152.40                  -35.00\n",
      "   10   4137  13S  14M C   N     101  103 -171.86 -125.60   49.01    1.57  -63.40 -164.36   37.47\n",
      "   10         14M  14M N   CA    103  104  156.68  140.50                  -40.50\n",
      "   11   4138  14M  15A C   N     109  111  -89.38  -68.20   42.27    4.05  -62.50  152.02   23.74\n",
      "   11         15A  15A N   CA    111  112  108.72  145.30                  -40.90\n",
      "   12   4139  15A  16A C   N     114  116  169.77 -134.00   63.01    1.44  -62.50 -167.76   38.09\n",
      "   12         16A  16A N   CA    116  117  175.43  147.00                  -40.90\n",
      "   13   4140  16A  17H C   N     119  121 -136.55 -125.60   67.93    2.36  -63.20  133.77   21.41\n",
      "   13         17H  17H N   CA    121  122 -154.16  138.80                  -42.30\n",
      "   14   4141  17H  18G C   N     129  131  -76.19  -80.20    4.78    0.35   82.20 -128.97    7.00\n",
      "   14         18G  18G N   CA    131  132  176.69  174.10                    8.50\n",
      "   15   4142  18G  19K C   N     133  135  -71.64  -70.20   14.31    1.07  -62.90  167.19   21.25\n",
      "   15         19K  19K N   CA    135  136  126.16  140.40                  -40.80\n",
      "   16   4177  53E  54S C   N     436  438 -136.45  -64.10   80.20    8.42  -64.10   80.20    8.42\n",
      "   16         54S  54S N   CA    438  439   -0.39  -35.00                  -35.00\n",
      "   17   4188  64N  65P C   N     521  523  -76.69  -64.50   13.49    1.09  -58.70  177.44   16.14\n",
      "   17         65P  65P N   CA    523  524  152.98  147.20                  -30.50\n",
      "   18   4241 117I 118V C   N     932  934  -56.39  -62.40   50.23    6.72  -62.40   50.23    6.72\n",
      "   18        118V 118V N   CA    934  935    7.47  -42.40                  -42.40\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    1   12   14   55   76  137  129  155  177  183  210\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      150\n",
      "Number of all, selected real atoms                :     1222    1222\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    13381   13381\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2608\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :         963.2237\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1205       0      0   0.007   0.007      17.632       1.000\n",
      " 2 Bond angle potential               :    1624       0      8   2.242   2.242      164.34       1.000\n",
      " 3 Stereochemical cosine torsion poten:     773       0     33  49.416  49.416      289.20       1.000\n",
      " 4 Stereochemical improper torsion pot:     506       0      0   1.479   1.479      23.834       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2608       2      2   0.008   0.008      18.277       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    2380       0      0   0.131   0.131      40.823       1.000\n",
      "10 Distance restraints 2 (N-O)        :    2535       0      0   0.178   0.178      74.743       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     146       0      2   4.473   4.473      34.453       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     125       0      2  67.065  67.065      25.521       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:      98       0      4  70.929  70.929      44.676       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      40       0      0  69.402  69.402      23.443       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      19       0      0  86.281  86.281      13.620       1.000\n",
      "18 Disulfide distance restraints      :       4       0      0   0.006   0.006     0.25204E-01   1.000\n",
      "19 Disulfide angle restraints         :       8       0      0   2.636   2.636      1.2271       1.000\n",
      "20 Disulfide dihedral angle restraints:       4       0      0  26.145  26.145      2.6144       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    1577       0      0   0.379   0.379      38.241       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     145      17     20  22.629  60.754      58.389       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     793       0      0   0.720   0.720      70.342       1.000\n",
      "27 Distance restraints 5 (X-Y)        :    1399       0      0   0.042   0.042      21.824       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: with_ligand.V99990002\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   15741.5723\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4125   1M   2K C   N       7    9  -81.14  -70.20   17.55    1.05  -62.90  166.10   22.61\n",
      "    1          2K   2K N   CA      9   10  154.11  140.40                  -40.80\n",
      "    2   4127   3V   4L C   N      23   25 -114.08 -108.50   17.45    0.87  -63.50  177.14   27.71\n",
      "    2          4L   4L N   CA     25   26  149.03  132.50                  -41.20\n",
      "    3   4128   4L   5L C   N      31   33  -70.68  -70.70   11.10    0.84  -63.50  166.25   23.50\n",
      "    3          5L   5L N   CA     33   34  152.70  141.60                  -41.20\n",
      "    4   4129   5L   6L C   N      39   41  -64.48  -70.70    6.54    0.52  -63.50  179.23   24.94\n",
      "    4          6L   6L N   CA     41   42  139.57  141.60                  -41.20\n",
      "    5   4130   6L   7L C   N      47   49  -99.59 -108.50    8.99    0.44  -63.50  178.57   22.98\n",
      "    5          7L   7L N   CA     49   50  133.68  132.50                  -41.20\n",
      "    6   4131   7L   8G C   N      55   57 -167.53 -167.20   47.77    1.88   82.20  161.74   12.32\n",
      "    6          8G   8G N   CA     57   58  126.83  174.60                    8.50\n",
      "    7   4132   8G   9F C   N      59   61 -175.75 -124.20   60.56    1.59  -63.20 -179.89   31.72\n",
      "    7          9F   9F N   CA     61   62  175.08  143.30                  -44.30\n",
      "    8   4134  10I  11F C   N      78   80  -83.49  -71.40   40.43    3.52  -63.20  147.82   19.20\n",
      "    8         11F  11F N   CA     80   81  102.12  140.70                  -44.30\n",
      "    9   4136  12C  13S C   N      95   97  -74.99  -72.40   16.62    0.89  -64.10  156.56   12.02\n",
      "    9         13S  13S N   CA     97   98  168.82  152.40                  -35.00\n",
      "   10   4137  13S  14M C   N     101  103  -82.45  -73.00   71.37    4.99  -63.40  114.36   15.69\n",
      "   10         14M  14M N   CA    103  104   72.26  143.00                  -40.50\n",
      "   11   4138  14M  15A C   N     109  111 -163.05 -134.00   29.64    0.78  -62.50 -165.70   37.20\n",
      "   11         15A  15A N   CA    111  112  152.85  147.00                  -40.90\n",
      "   12   4139  15A  16A C   N     114  116  -64.85  -68.20    7.53    0.50  -62.50  179.47   29.30\n",
      "   12         16A  16A N   CA    116  117  138.56  145.30                  -40.90\n",
      "   13   4140  16A  17H C   N     119  121 -136.06 -125.60   11.58    0.49  -63.20 -169.40   20.48\n",
      "   13         17H  17H N   CA    121  122  133.82  138.80                  -42.30\n",
      "   14   4141  17H  18G C   N     129  131 -176.17 -167.20   19.38    0.79   82.20 -179.71   13.15\n",
      "   14         18G  18G N   CA    131  132  157.42  174.60                    8.50\n",
      "   15   4142  18G  19K C   N     133  135  -81.94  -70.20   17.34    1.45  -62.90  169.51   21.04\n",
      "   15         19K  19K N   CA    135  136  127.63  140.40                  -40.80\n",
      "   16   4177  53E  54S C   N     436  438 -134.90  -64.10   74.64    8.43  -64.10   74.64    8.43\n",
      "   16         54S  54S N   CA    438  439  -11.35  -35.00                  -35.00\n",
      "   17   4188  64N  65P C   N     521  523  -76.19  -64.50   14.43    1.02  -58.70  174.73   15.88\n",
      "   17         65P  65P N   CA    523  524  155.65  147.20                  -30.50\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    0    0    1    8   13   58   85  144  124  137  193  197  192\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "with_ligand.B99990001.pdb      986.91687\n",
      "with_ligand.B99990002.pdb      963.22369\n",
      "\n"
     ]
    }
   ],
   "source": [
    "alignm2.salign()\n",
    "alignm2.write(file='all_in_one_ligand.ali', alignment_format='PIR')\n",
    "s = alignm2[0]\n",
    "pdb = alignm2[1]\n",
    "\n",
    "print s.code, pdb.code\n",
    "\n",
    "## Создаем объект automodel\n",
    "a = modeller.automodel.automodel(env, alnfile='all_in_one_ligand.ali', knowns= pdb.code , sequence = s.code )\n",
    "\n",
    "a.name='mod'+s.code\n",
    "a.starting_model = 1\n",
    "a.ending_model = 2\n",
    "a.make()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "a0c369ee92ef4e68974345123186bd32",
       "version_major": 2,
       "version_minor": 0
      },
      "text/html": [
       "<p>Failed to display Jupyter Widget of type <code>NGLWidget</code>.</p>\n",
       "<p>\n",
       "  If you're reading this message in Jupyter Notebook or JupyterLab, it may mean\n",
       "  that the widgets JavaScript is still loading. If this message persists, it\n",
       "  likely means that the widgets JavaScript library is either not installed or\n",
       "  not enabled. See the <a href=\"https://ipywidgets.readthedocs.io/en/stable/user_install.html\">Jupyter\n",
       "  Widgets Documentation</a> for setup instructions.\n",
       "</p>\n",
       "<p>\n",
       "  If you're reading this message in another notebook frontend (for example, a static\n",
       "  rendering on GitHub or <a href=\"https://nbviewer.jupyter.org/\">NBViewer</a>),\n",
       "  it may mean that your frontend doesn't currently support widgets.\n",
       "</p>\n"
      ],
      "text/plain": [
       "NGLWidget()"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "nglview.show_structure_file('with_ligand.B99990001.pdb')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Restrains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 67,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['R    3   1   1   1   2   2   1   100    99       1.4200    0.0263 \\n', 'R    3   1   1   1   2   2   1    97    98       1.4300    0.0304 \\n', 'R    3   1   1   1   2   2   1   102   101       1.2300    0.0218 \\n', 'R    3   1   1   1   2   2   1   101    98       1.4900    0.0344 \\n', 'R    3   1   1   1   2   2   1   101   103       1.3450    0.0282 \\n', 'R    3   1   1   1   2   2   1   105   104       1.5380    0.0364 \\n', 'R    3   1   1   1   2   2   1   106   105       1.5300    0.0364 \\n', 'R    3   1   1   1   2   2   1   107   106       1.8180    0.0386 \\n', 'R    3   1   1   1   2   2   1   108   107       1.8160    0.0351 \\n', 'R    3   1   1   1   2   2   1   103   104       1.4300    0.0304 \\n', 'R    3   1   1   1   2   2   1   110   109       1.2300    0.0218 \\n', 'R    3   1   1   1   2   2   1   109   104       1.4900    0.0344 \\n', 'R    3   1   1   1   2   2   1   109   111       1.3450    0.0282 \\n', 'R    3   1   1   1   2   2   1   113   112       1.5380    0.0364 \\n', 'R    3   1   1   1   2   2   1   111   112       1.4300    0.0304 \\n', 'R    3   1   1   1   2   2   1   115   114       1.2300    0.0218 \\n', 'R    3   1   1   1   2   2   1   114   112       1.4900    0.0344 \\n', 'R    3   1   1   1   2   2   1   114   116       1.3450    0.0282 \\n', 'R    3   1   1   1   2   2   1   118   117       1.5380    0.0364 \\n']\n"
     ]
    }
   ],
   "source": [
    "with open('with_ligand.rsr') as fn:\n",
    "    lines = fn.readlines()\n",
    "print(lines[100:119])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 72,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "automodel__W> Topology and/or parameter libraries already in memory. These will\n",
      "                be used instead of the automodel defaults. If this is not what you\n",
      "                want, clear them before creating the automodel object with\n",
      "                env.libs.topology.clear() and env.libs.parameters.clear()\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "\n",
      "check_ali___> Checking the sequence-structure alignment. \n",
      "\n",
      "Implied intrachain target CA(i)-CA(i+1) distances longer than  8.0 angstroms:\n",
      "\n",
      "ALN_POS  TMPL  RID1  RID2  NAM1  NAM2     DIST\n",
      "----------------------------------------------\n",
      "END OF TABLE\n",
      "\n",
      "getf_______W> RTF restraint not found in the atoms list:\n",
      "              residue type, indices:    10   147\n",
      "              atom names           : C     +N\n",
      "              atom indices         :  1177     0\n",
      "\n",
      "getf_______W> RTF restraint not found in the atoms list:\n",
      "              residue type, indices:    10   147\n",
      "              atom names           : C     CA    +N    O\n",
      "              atom indices         :  1177  1172     0  1178\n",
      "fndatmi_285W> Only      129 residues out of      132 contain atoms of type  CA\n",
      "              (This is usually caused by non-standard residues, such\n",
      "              as ligands, or by PDB files with missing atoms.)\n",
      "patch_s_522_> Number of disulfides patched in MODEL:        4\n",
      "mdtrsr__446W> A potential that relies on one protein is used, yet you have at\n",
      "              least one known structure available. MDT, not library, potential is used.\n",
      "iup2crm_280W> No topology library in memory or assigning a BLK residue.\n",
      "              Default CHARMM atom type assigned:  C1 -->  CT2\n",
      "              This message is written only for the first such atom.\n",
      "43 atoms in HETATM/BLK residues constrained\n",
      "to protein atoms within 2.30 angstroms\n",
      "and protein CA atoms within 10.00 angstroms\n",
      "43 atoms in residues without defined topology\n",
      "constrained to be rigid bodies\n",
      "runcmd_____W>: creation of new member 'my_data' in <Model containing 1 chain, 150 residues, and 1222 atoms>: possible typo!\n",
      "condens_443_> Restraints marked for deletion were removed.\n",
      "              Total number of restraints before, now:    14414    13382\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      150\n",
      "Number of all, selected real atoms                :     1222    1222\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    13382   13382\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2585\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        2246.6326\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1205       0      0   0.007   0.007      17.858       1.000\n",
      " 2 Bond angle potential               :    1624       0      7   2.158   2.158      151.16       1.000\n",
      " 3 Stereochemical cosine torsion poten:     773       0     36  49.119  49.119      285.18       1.000\n",
      " 4 Stereochemical improper torsion pot:     506       0      0   1.369   1.369      21.554       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2585       2      3   0.012   0.012      43.782       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    2380       0      0   0.130   0.130      41.306       1.000\n",
      "10 Distance restraints 2 (N-O)        :    2535       0      0   0.182   0.182      78.892       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     146       0      1   4.305   4.305      31.904       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     125       0      1  70.547  70.547      33.279       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:      98       0      0  76.072  76.072      44.232       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      40       0      0  78.179  78.179      28.371       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      19       0      0  74.354  74.354      11.022       1.000\n",
      "18 Disulfide distance restraints      :       4       0      0   0.007   0.007     0.37925E-01   1.000\n",
      "19 Disulfide angle restraints         :       8       0      0   2.884   2.884      1.4697       1.000\n",
      "20 Disulfide dihedral angle restraints:       4       0      0  27.393  27.393      2.7304       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    1577       0      0   0.394   0.394      34.870       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     145      17     17  22.542  60.192      57.289       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     793       0      1   0.729   0.729      61.817       1.000\n",
      "27 Distance restraints 5 (X-Y)        :    1400       3     17   0.194   0.194      1299.9       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: with_ligand.V99990001\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   17283.4844\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4125   1M   2K C   N       7    9  -91.11  -70.20   24.57    1.51  -62.90  168.29   23.49\n",
      "    1          2K   2K N   CA      9   10  153.29  140.40                  -40.80\n",
      "    2   4127   3V   4L C   N      23   25  -66.28  -70.70    4.71    0.37  -63.50  178.88   25.00\n",
      "    2          4L   4L N   CA     25   26  139.95  141.60                  -41.20\n",
      "    3   4128   4L   5L C   N      31   33 -122.63 -108.50   29.61    1.42  -63.50  170.83   27.38\n",
      "    3          5L   5L N   CA     33   34  158.52  132.50                  -41.20\n",
      "    4   4129   5L   6L C   N      39   41  -72.71  -70.70   26.69    1.95  -63.50  150.87   21.49\n",
      "    4          6L   6L N   CA     41   42  168.21  141.60                  -41.20\n",
      "    5   4130   6L   7L C   N      47   49  -60.44  -70.70   10.82    0.86  -63.50  179.39   25.08\n",
      "    5          7L   7L N   CA     49   50  138.16  141.60                  -41.20\n",
      "    6   4131   7L   8G C   N      55   57   95.69   78.70   44.61    0.69   82.20  144.77    7.70\n",
      "    6          8G   8G N   CA     57   58  152.65 -166.10                    8.50\n",
      "    7   4132   8G   9F C   N      59   61  -95.76  -71.40   59.84    5.28  -63.20  134.36   16.61\n",
      "    7          9F   9F N   CA     61   62   86.05  140.70                  -44.30\n",
      "    8   4134  10I  11F C   N      78   80  -80.76  -71.40   12.85    0.74  -63.20  167.12   24.39\n",
      "    8         11F  11F N   CA     80   81  149.51  140.70                  -44.30\n",
      "    9   4136  12C  13S C   N      95   97 -168.96 -136.60   44.58    1.42  -64.10  177.43   20.42\n",
      "    9         13S  13S N   CA     97   98 -178.13  151.20                  -35.00\n",
      "   10   4137  13S  14M C   N     101  103  -87.69  -73.00   40.93    3.03  -63.40  147.31   20.22\n",
      "   10         14M  14M N   CA    103  104  104.79  143.00                  -40.50\n",
      "   11   4138  14M  15A C   N     109  111 -161.80 -134.00   29.34    0.69  -62.50 -169.36   36.53\n",
      "   11         15A  15A N   CA    111  112  156.36  147.00                  -40.90\n",
      "   12   4139  15A  16A C   N     114  116 -161.14 -134.00   30.58    0.70  -62.50 -173.74   35.77\n",
      "   12         16A  16A N   CA    116  117  161.10  147.00                  -40.90\n",
      "   13   4140  16A  17H C   N     119  121  -71.21  -67.60   10.82    0.72  -63.20  167.68   21.47\n",
      "   13         17H  17H N   CA    121  122  150.21  140.00                  -42.30\n",
      "   14   4141  17H  18G C   N     129  131  157.56 -167.20   35.71    0.83   82.20  177.16   12.06\n",
      "   14         18G  18G N   CA    131  132  168.84  174.60                    8.50\n",
      "   15   4142  18G  19K C   N     133  135 -118.65 -118.00   10.45    0.50  -62.90  178.59   26.51\n",
      "   15         19K  19K N   CA    135  136  149.54  139.10                  -40.80\n",
      "   16   4177  53E  54S C   N     436  438 -137.19  -64.10   78.62    8.61  -64.10   78.62    8.61\n",
      "   16         54S  54S N   CA    438  439   -6.04  -35.00                  -35.00\n",
      "   17   4188  64N  65P C   N     521  523  -71.92  -64.50   12.82    1.34  -58.70  167.77   12.82\n",
      "   17         65P  65P N   CA    523  524  136.75  147.20                  -30.50\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 27                           : Distance restraints 5 (X-Y)             \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13382 148. 150. C1  O1L  1180 1222    9.16    3.00    6.16   61.56    3.00    6.16   61.56\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:  1180 1197   2.294\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    1    0    2    6    9   59   92  121  131  144  188  197  205\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "iupac_m_397W> Atoms were not swapped because of the uncertainty of how to handle the H atom.\n",
      "\n",
      "\n",
      ">> ENERGY; Differences between the model's features and restraints:\n",
      "Number of all residues in MODEL                   :      150\n",
      "Number of all, selected real atoms                :     1222    1222\n",
      "Number of all, selected pseudo atoms              :        0       0\n",
      "Number of all static, selected restraints         :    13382   13382\n",
      "COVALENT_CYS                                      :        F\n",
      "NONBONDED_SEL_ATOMS                               :        1\n",
      "Number of non-bonded pairs (excluding 1-2,1-3,1-4):     2761\n",
      "Dynamic pairs routine                             : 2, NATM x NATM cell sorting\n",
      "Atomic shift for contacts update (UPDATE_DYNAMIC) :    0.390\n",
      "LENNARD_JONES_SWITCH                              :    6.500   7.500\n",
      "COULOMB_JONES_SWITCH                              :    6.500   7.500\n",
      "RESIDUE_SPAN_RANGE                                :        0   99999\n",
      "NLOGN_USE                                         :       15\n",
      "CONTACT_SHELL                                     :    4.000\n",
      "DYNAMIC_PAIRS,_SPHERE,_COULOMB,_LENNARD,_MODELLER :        T       T       F       F       F\n",
      "SPHERE_STDV                                       :    0.050\n",
      "RADII_FACTOR                                      :    0.820\n",
      "Current energy                                    :        2485.3855\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "\n",
      "Summary of the restraint violations: \n",
      "\n",
      "   NUM     ... number of restraints.\n",
      "   NUMVI   ... number of restraints with RVIOL > VIOL_REPORT_CUT[i].\n",
      "   RVIOL   ... relative difference from the best value.\n",
      "   NUMVP   ... number of restraints with -Ln(pdf) > VIOL_REPORT_CUT2[i].\n",
      "   RMS_1   ... RMS(feature, minimally_violated_basis_restraint, NUMB).\n",
      "   RMS_2   ... RMS(feature, best_value, NUMB).\n",
      "   MOL.PDF ... scaled contribution to -Ln(Molecular pdf).\n",
      "\n",
      " #                     RESTRAINT_GROUP      NUM   NUMVI  NUMVP   RMS_1   RMS_2         MOL.PDF     S_i\n",
      "------------------------------------------------------------------------------------------------------\n",
      " 1 Bond length potential              :    1205       0      0   0.008   0.008      20.348       1.000\n",
      " 2 Bond angle potential               :    1624       1     13   2.369   2.369      184.07       1.000\n",
      " 3 Stereochemical cosine torsion poten:     773       0     39  49.523  49.523      298.82       1.000\n",
      " 4 Stereochemical improper torsion pot:     506       0      0   1.354   1.354      21.402       1.000\n",
      " 5 Soft-sphere overlap restraints     :    2761       2      3   0.012   0.012      44.627       1.000\n",
      " 6 Lennard-Jones 6-12 potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 7 Coulomb point-point electrostatic p:       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 8 H-bonding potential                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      " 9 Distance restraints 1 (CA-CA)      :    2380       0      0   0.146   0.146      57.730       1.000\n",
      "10 Distance restraints 2 (N-O)        :    2535       5     21   0.265   0.265      183.61       1.000\n",
      "11 Mainchain Phi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "12 Mainchain Psi dihedral restraints  :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "13 Mainchain Omega dihedral restraints:     146       0      4   4.999   4.999      43.024       1.000\n",
      "14 Sidechain Chi_1 dihedral restraints:     125       0      1  69.404  69.404      34.735       1.000\n",
      "15 Sidechain Chi_2 dihedral restraints:      98       0      1  71.129  71.129      42.556       1.000\n",
      "16 Sidechain Chi_3 dihedral restraints:      40       0      0  81.587  81.587      27.545       1.000\n",
      "17 Sidechain Chi_4 dihedral restraints:      19       0      0 105.660 105.660      15.359       1.000\n",
      "18 Disulfide distance restraints      :       4       0      0   0.008   0.008     0.43987E-01   1.000\n",
      "19 Disulfide angle restraints         :       8       0      0   3.443   3.443      2.0943       1.000\n",
      "20 Disulfide dihedral angle restraints:       4       0      0  24.596  24.596      2.0601       1.000\n",
      "21 Lower bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "22 Upper bound distance restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "23 Distance restraints 3 (SDCH-MNCH)  :    1577       0      0   0.412   0.412      39.042       1.000\n",
      "24 Sidechain Chi_5 dihedral restraints:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "25 Phi/Psi pair of dihedral restraints:     145      21     19  26.783  65.931      86.711       1.000\n",
      "26 Distance restraints 4 (SDCH-SDCH)  :     793       0      4   0.853   0.853      86.460       1.000\n",
      "27 Distance restraints 5 (X-Y)        :    1400       3     16   0.196   0.196      1295.1       1.000\n",
      "28 NMR distance restraints 6 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "29 NMR distance restraints 7 (X-Y)    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "30 Minimal distance restraints        :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "31 Non-bonded restraints              :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "32 Atomic accessibility restraints    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "33 Atomic density restraints          :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "34 Absolute position restraints       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "35 Dihedral angle difference restraint:       0       0      0   0.000   0.000      0.0000       1.000\n",
      "36 GBSA implicit solvent potential    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "37 EM density fitting potential       :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "38 SAXS restraints                    :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "39 Symmetry restraints                :       0       0      0   0.000   0.000      0.0000       1.000\n",
      "\n",
      "\n",
      "\n",
      "# Heavy relative violation of each residue is written to: with_ligand.V99990002\n",
      "# The profile is NOT normalized by the number of restraints.\n",
      "# The profiles are smoothed over a window of residues:    1\n",
      "# The sum of all numbers in the file:   18306.4609\n",
      "\n",
      "\n",
      "\n",
      "List of the violated restraints:\n",
      "   A restraint is violated when the relative difference\n",
      "   from the best value (RVIOL) is larger than CUTOFF.\n",
      "\n",
      "   ICSR   ... index of a restraint in the current set.\n",
      "   RESNO  ... residue numbers of the first two atoms.\n",
      "   ATM    ... IUPAC atom names of the first two atoms.\n",
      "   FEAT   ... the value of the feature in the model.\n",
      "   restr  ... the mean of the basis restraint with the smallest\n",
      "              difference from the model (local minimum).\n",
      "   viol   ... difference from the local minimum.\n",
      "   rviol  ... relative difference from the local minimum.\n",
      "   RESTR  ... the best value (global minimum).\n",
      "   VIOL   ... difference from the best value.\n",
      "   RVIOL  ... relative difference from the best value.\n",
      "\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature  2                           : Bond angle potential                    \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   1419  21M  21M N   CA    155  156  123.55  107.00   16.55    4.76  107.00   16.55    4.76\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 10                           : Distance restraints 2 (N-O)             \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   7797  54S  21M N   O     438  162   12.01   10.11    1.89    4.81   10.11    1.89    4.81\n",
      "    2   7848  56F  21M N   O     448  162    7.63    5.56    2.07    5.35    5.56    2.07    5.35\n",
      "    3   8085  69S  89P N   O     551  719    8.61    6.49    2.12    4.67    6.49    2.12    4.67\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 25                           : Phi/Psi pair of dihedral restraints     \n",
      "List of the RVIOL violations larger than   :       6.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1   4125   1M   2K C   N       7    9   62.00   56.60   12.53    0.63  -62.90  142.25   24.85\n",
      "    1          2K   2K N   CA      9   10   27.29   38.60                  -40.80\n",
      "    2   4127   3V   4L C   N      23   25 -101.55 -108.50    8.67    0.38  -63.50  172.76   22.11\n",
      "    2          4L   4L N   CA     25   26  127.32  132.50                  -41.20\n",
      "    3   4128   4L   5L C   N      31   33 -113.74 -108.50   17.03    0.86  -63.50  177.36   27.72\n",
      "    3          5L   5L N   CA     33   34  148.71  132.50                  -41.20\n",
      "    4   4129   5L   6L C   N      39   41  -59.67  -70.70   34.87    2.27  -63.50  149.77   21.02\n",
      "    4          6L   6L N   CA     41   42  108.52  141.60                  -41.20\n",
      "    5   4130   6L   7L C   N      47   49 -136.74 -108.50   31.76    1.40  -63.50 -173.28   30.46\n",
      "    5          7L   7L N   CA     49   50  147.04  132.50                  -41.20\n",
      "    6   4131   7L   8G C   N      55   57 -120.55  -80.20   53.21    3.83   82.20 -155.39   15.90\n",
      "    6          8G   8G N   CA     57   58  139.41  174.10                    8.50\n",
      "    7   4132   8G   9F C   N      59   61 -123.05 -124.20    7.86    0.34  -63.20 -170.47   22.71\n",
      "    7          9F   9F N   CA     61   62  135.53  143.30                  -44.30\n",
      "    8   4134  10I  11F C   N      78   80  -93.35  -71.40   44.11    3.92  -63.20  149.81   18.88\n",
      "    8         11F  11F N   CA     80   81  102.44  140.70                  -44.30\n",
      "    9   4136  12C  13S C   N      95   97 -147.16 -136.60   28.25    1.17  -64.10  169.37   18.25\n",
      "    9         13S  13S N   CA     97   98  177.39  151.20                  -35.00\n",
      "   10   4137  13S  14M C   N     101  103  -87.52  -73.00   55.55    4.01  -63.40  132.10   17.98\n",
      "   10         14M  14M N   CA    103  104   89.38  143.00                  -40.50\n",
      "   11   4138  14M  15A C   N     109  111  179.54 -134.00   51.08    1.16  -62.50 -168.49   37.55\n",
      "   11         15A  15A N   CA    111  112  168.23  147.00                  -40.90\n",
      "   12   4139  15A  16A C   N     114  116  -96.23  -68.20   42.27    4.15  -62.50  158.19   24.50\n",
      "   12         16A  16A N   CA    116  117  113.66  145.30                  -40.90\n",
      "   13   4140  16A  17H C   N     119  121 -122.66 -125.60   27.42    0.98  -63.20  164.93   17.81\n",
      "   13         17H  17H N   CA    121  122  111.54  138.80                  -42.30\n",
      "   14   4142  18G  19K C   N     133  135 -122.18 -118.00    4.56    0.21  -62.90 -172.31   21.90\n",
      "   14         19K  19K N   CA    135  136  137.28  139.10                  -40.80\n",
      "   15   4144  20R  21M C   N     153  155  -68.70  -63.40    6.53    0.91  -73.00  179.75   12.07\n",
      "   15         21M  21M N   CA    155  156  -36.70  -40.50                  143.00\n",
      "   16   4145  21M  22E C   N     161  163   68.75  -69.30  140.05    9.76  -69.30  140.05    9.76\n",
      "   16         22E  22E N   CA    163  164  118.89  142.50                  142.50\n",
      "   17   4177  53E  54S C   N     436  438 -136.64  -64.10   77.51    8.57  -64.10   77.51    8.57\n",
      "   17         54S  54S N   CA    438  439   -7.69  -35.00                  -35.00\n",
      "   18   4188  64N  65P C   N     521  523  -68.71  -64.50   67.27    4.80  -58.70  115.59   10.37\n",
      "   18         65P  65P N   CA    523  524 -145.66  147.20                  -30.50\n",
      "   19   4189  65P  66G C   N     528  530  -52.61  -62.40   18.52    3.29   82.20  139.03   10.34\n",
      "   19         66G  66G N   CA    530  531  -25.48  -41.20                    8.50\n",
      "   20   4212  88T  89P C   N     711  713  -43.58  -58.70   62.45    4.07  -64.50  123.49   10.35\n",
      "   20         89P  89P N   CA    713  714  -91.10  -30.50                  147.20\n",
      "   21   4213  89P  90H C   N     718  720  -99.99 -125.60   60.30    1.76   56.30  162.20   22.66\n",
      "   21         90H  90H N   CA    720  721   84.20  138.80                   40.80\n",
      "\n",
      "-------------------------------------------------------------------------------------------------\n",
      "\n",
      "Feature 27                           : Distance restraints 5 (X-Y)             \n",
      "List of the RVIOL violations larger than   :       4.5000\n",
      "\n",
      "    #   ICSR  RESNO1/2 ATM1/2   INDATM1/2    FEAT   restr    viol   rviol   RESTR    VIOL   RVIOL\n",
      "    1  13382 148. 150. C1  O1L  1180 1222    9.10    3.00    6.10   61.04    3.00    6.10   61.04\n",
      "\n",
      "\n",
      "report______> Distribution of short non-bonded contacts:\n",
      "\n",
      "  serious non-bonded atom clash:  1180 1197   2.297\n",
      "\n",
      "DISTANCE1:  0.00 2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40\n",
      "DISTANCE2:  2.10 2.20 2.30 2.40 2.50 2.60 2.70 2.80 2.90 3.00 3.10 3.20 3.30 3.40 3.50\n",
      "FREQUENCY:     0    0    1    0    2   10   15   77   95  146  127  140  181  227  213\n",
      "\n",
      "\n",
      "<< end of ENERGY.\n",
      "\n",
      ">> Summary of successfully produced models:\n",
      "Filename                          molpdf\n",
      "----------------------------------------\n",
      "with_ligand.B99990001.pdb     2246.63257\n",
      "with_ligand.B99990002.pdb     2485.38550\n",
      "\n"
     ]
    }
   ],
   "source": [
    "class mymodel(modeller.automodel.automodel):\n",
    "    def special_restraints(self, aln):\n",
    "        rsr = self.restraints\n",
    "        at = self.atoms\n",
    "        for x,y in [('C1:148','O1L:150')]:\n",
    "            self.my_data = modeller.forms.gaussian(group=modeller.physical.xy_distance, \n",
    "                                        feature=modeller.features.distance(\n",
    "                                        at[x],at[y]),mean=3.0, stdev=0.1)\n",
    "            rsr.add(modeller.forms.gaussian(group=modeller.physical.xy_distance, \n",
    "                                        feature=modeller.features.distance(\n",
    "                                        at[x],at[y]),mean=3.0, stdev=0.1))\n",
    "\n",
    "from modeller import *\n",
    "from modeller.automodel import *    \n",
    "a = mymodel(env, alnfile='all_in_one_ligand.ali',\n",
    "               knowns= pdb.code , sequence = alignm2[0].code)\n",
    "a.starting_model = 1\n",
    "a.ending_model = 2\n",
    "a.make()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#### <a href='Tikhonova. HW6.ipynb'> Download this notebook </a>"
   ]
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "hse-py27",
   "language": "python",
   "name": "hse"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
